tests/periph_spi/tests/04__periph_spi_timer.robot

```diff
@@ -13,66 +13,88 @@ Test Teardown       Run Keywords    SPI Release Should Succeed
 
 Resource            periph_spi.keywords.txt
 Resource            api_shell.keywords.txt
+Resource            util.keywords.txt
 
 Variables           test_vars.py
 
 Force Tags          periph  spi
 
 
 *** Keywords ***
-SPI Clock Speed Check
+SPI Clock Speed Check Setup
     [Documentation]     Main routine to verify the clock speed
     [Arguments]         ${clk_speed_string}  ${clk_speed_value}  ${size}
+    Set Test Variable                   ${size}
     PHiLIP.write and execute            spi.mode.if_type    3
-    SPI Acquire Should Succeed          0  ${clk_speed_string}
-    SPI Transfer Bytes Should Succeed   cont=0   in_len=${size}
-    API Call Should Succeed             PHiLIP.Read Reg         sys.sys_clk
-    ${sys_clk} =  Set Variable          ${RESULT['data']}
-    API Call Should Succeed             PHiLIP.Read Reg         spi.byte_ticks
-    ${spi_speed_limits} =  Set Variable  
-    ...  ${spi_speed_limits(${clk_speed_value}, ${sys_clk}, ${size})}
-    Should Be True  
-    ...  ${spi_speed_limits}[0] < ${${RESULT['data']} * ${size}} and ${spi_speed_limits}[1] > ${${RESULT['data']} * ${size}}
-    ...  Time should be between ${convert_ticks_to_us(${spi_speed_limits[0]},${sys_clk})}us and ${convert_ticks_to_us(${spi_speed_limits[1]},${sys_clk})}us but was ${convert_ticks_to_us(${RESULT['data']} * ${size}, ${sys_clk})}us
+    SPI Acquire Should Succeed          0                   ${clk_speed_string}
+    SPI Transfer Bytes Should Succeed   cont=0              in_len=${size}
+    API Call Should Succeed             PHiLIP.Read Reg     sys.sys_clk
+    Set Test Variable                   ${sys_clk}          ${RESULT['data']}
+    API Call Should Succeed             PHiLIP.Read Reg     spi.byte_ticks
+    Set Test Variable                   ${byte_ticks}       ${RESULT['data']}
+    Set Test Variable                   ${spi_speed_limits}
+    ...                                   ${spi_speed_limits(${clk_speed_value}, ${sys_clk}, ${size})}
+    Set Test Variable                   ${lower_time_limit}
+    ...                                   ${ticks_to_us(${spi_speed_limits[0]}, ${sys_clk})}
+    Set Test Variable                   ${upper_time_limit}
+    ...                                   ${ticks_to_us(${spi_speed_limits[1]}, ${sys_clk})}
+    Set Test Variable                   ${measured_time}
+    ...                                   ${ticks_to_us((${byte_ticks} * ${size}), ${sys_clk})}
+    Run Keyword And Warn On Failure     Compare Time Measures
+
+    Record Property  Measured time ${clk_speed_string}Hz ${size}Byte  ${measured_time}us
```

Comment:
This should be clock frequency.

So something like:
```
Record Property              freq=${clk_speed_string}_size=${size}  ${get_spi_clock_stats(clk_speed_value, size, measured_time}
```

And then implement the get_spi_clock_stats function somewhere:
```py
def get_spi_clock_stats(clk_speed_value, size, measured_time):
    clk_speed_value = int(clk_speed_value)
    size = int(size)
    measured_time = int(measured_time)
    result = {}
    result['expected'] = clk_speed_value # Say 100000
    result['measured'] = int(1000000 / measured_time / size) # Say 125000 due to hardware constraints
    return result
```

Note that the measured is in clock frequency and normalized for the size of bytes sent.
I don't think we need to include the limits here as the pass/warn/fail should indicate limits.  If we were graph this we should be able to see trends or jumps. Eventually we will add a calculated field too.

Also the parsing is already done so if we were to operate on the data we wouldn't have to strip out the units (ie. us or Hz) and cast it to ints since we do that before hand.